src/ResourceManager/Websites/Commands.Websites/Cmdlets/BackupRestore/RestoreAzureWebAppSnapshot.cs

```diff
@@ -48,23 +48,23 @@ public class RestoreAzureWebAppSnapshot : WebAppOptionalSlotBaseCmdlet
         public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
-            Site targetApp = WebsitesClient.GetWebApp(ResourceGroupName, Name, Slot);
-            SnapshotRecoveryTarget target = new SnapshotRecoveryTarget()
+            Site sourceApp = WebsitesClient.GetWebApp(InputObject.ResourceGroupName, InputObject.Name, InputObject.Slot);
```

Comment:
Why are you changing these to InputObject.<>? Shouldn't this be using the parameters for retrieving the web app rather than the snapshot?

Reply:
This is because the API is different now. In previous API versions you would make a recovery request to the app where the snapshot was taken (the source app) and specify which app to restore the snapshot to (the target app) in the request body. The semantics have been switched around. Now you send a RestoreSnapshot request to the target app, and specify the source app inside the request body. The API was changed to fix some scenarios in ARM.

InputObject is the source snapshot, and the *Name parameters are for the target site name. I swapped the code around so the cmdlet will still work the same way it used to, but internally the API flow will be fixed. I hope this won't be considered a breaking change since as far as the end user is concerned, the cmdlet works the same way.

Reply:
That makes sense to me, thanks for the clarification.  That will not be considered a breaking change.